File: PersonifiBackend/src/PersonifiBackend.Core/Entities/Bucket.cs
namespace PersonifiBackend.Core.Entities;

public class Bucket
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Color { get; set; }
    public decimal CurrentBalance { get; set; }
    public decimal? TargetAmount { get; set; }

    public int AccountId { get; set; }
    public Account Account { get; set; } = null!;
}

File: personifi-app/src/types/bucket.ts
export interface BucketDto {
  id: number;
  name: string;
  color?: string;
  currentBalance: number;
  targetAmount?: number;
}

export interface CreateBucketDto {
  name: string;
  color?: string;
  targetAmount?: number;
  currentBalance?: number;
}

export interface UpdateBucketDto {
  name: string;
  color?: string;
  targetAmount?: number;
  currentBalance?: number;
}
